application/models/SurveysGroups.php

```diff
@@ -421,20 +421,22 @@ protected static function getPermissionCriteria()
         $criteriaPerm = new CDbCriteria();
         if (!Permission::model()->hasGlobalPermission("surveys", 'read') || !Permission::model()->hasGlobalPermission("surveysgroups", 'read')) {
             $userid = App()->getCurrentUserId();
-            /* owner of surveygroup */
-            $criteriaPerm->compare('t.owner_id', $userid, false);
-            /* Simple permission on SurveysGroup inside a group */
-            $criteriaPerm->mergeWith(array(
-                'join' => "LEFT JOIN {{permissions}} AS permissions ON (permissions.entity_id = t.gsid AND permissions.permission='group' AND permissions.entity='surveysgroups' AND permissions.uid='" . $userid . "') ",
-            ));
-            $criteriaPerm->compare('permissions.read_p', '1', false, 'OR');
-            /* Permission on Survey inside a group */
-            $criteriaPerm->mergeWith(array(
-                'join' => "LEFT JOIN {{surveys}} AS surveys ON (surveys.gsid = t.gsid)
-                        LEFT JOIN {{permissions}} AS surveypermissions ON (surveypermissions.entity_id = surveys.sid AND surveypermissions.permission='survey' AND surveypermissions.entity='survey' AND surveypermissions.uid='" . $userid . "') ",
-            ));
-            $criteriaPerm->compare('surveys.owner_id', $userid, false, 'OR');
-            $criteriaPerm->compare('surveypermissions.read_p', '1', false, 'OR');
+            if (!empty($userid)) {
```

Comment:
No user id when listing SurveysGroups ? When this happens ?  CLI/console ?

`Permission::model()->hasGlobalPermission("surveysgroups", 'read') `return false, then maybe we need to disallow view any group ?

Reply:
:)